### PR TITLE
Update `prefer_null_aware_operators` for null safety; migrate tests

### PR DESCRIPTION
**Pull request by srawlins**
# Description

This updates the check to support `!` operators as well, to fix the referenced bug.

Also all of the tests are migrated.

Fixes https://github.com/dart-lang/linter/issues/4646

### CODE REVIEW

File: lib/src/rules/prefer_null_aware_operators.dart [hunk 1]
```diff
@@ -59,46 +59,48 @@ class _Visitor extends SimpleAstVisitor {
...
-          return;
-        }
-      } else {
-        if (node.elseExpression is! NullLiteral) {
-          return;
-        }
-      }
+    if (condition is! BinaryExpression) {
+      return;
+    }
```
Comment (pq) [inline lib/src/rules/prefer_null_aware_operators.dart:64]: Nice one.

**Review by pq — APPROVED** [review]
Sweet!

**Comment by srawlins** [comment]
Don't merge this yet. ONE failure in [flutter/flutter](https://logs.chromium.org/logs/dart/buildbucket/cr-buildbucket/8773355066382611393/+/u/analyze_flutter_flutter/stdout?format=raw).

**Comment by srawlins** [comment]
Fixed in flutter.

**Pull request merged**